lib/chef/knife/azurerm_base.rb

```diff
@@ -109,8 +109,21 @@ def token_details_for_linux
       def check_token_validity(token_details)
         time_difference = Time.parse(token_details[:expiry_time]) - Time.now.utc
         if time_difference <= 0
-          raise "Token has expired. Please run any XPLAT command like 'azure vm list' to get new token OR run 'azure login' command"
+          begin
+            xplat_command = Mixlib::ShellOut.new("azure vm image list-skus westus RedHat RHEL").run_command
+          rescue Exception
```

Comment:
Can we raise the exception here `Token has expired. Please run 'azure login' command` and exit? If the `azure vm image list-skus` command fails, then `token_details` data will remain same for both linux and windows and the `time_difference` in the code below will still be less than 0.
Please correct me if I am missing something here.